wagtail/images/views/images.py

```diff
@@ -111,12 +111,15 @@ def get_next_url(self):
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
 
+        view_mode = self.request.GET.get("view", "grid")
+
         context.update(
             {
                 "next": self.get_next_url(),
                 "current_collection": self.current_collection,
                 "current_ordering": self.ordering,
                 "ORDERING_OPTIONS": self.ORDERING_OPTIONS,
+                "view_mode": view_mode,
```

Comment:
Need tests that confirm:

- the view mode is passed correctly
- the listings’ HTML are adapted based on the view mode

From Sage: need tests that would fail if they ran against the previous implementation of those listings (so demonstrate the changes add this new behavior).

---

Assert contents of title column with title + filename (below title), matching the test’s created image

Reply:
The test client is a Python class that acts as a dummy web browser, allowing you to test your views and interact with your Django-powered application programmatically. It comes with html asserts, on of them:
https://docs.djangoproject.com/en/5.2/topics/testing/tools/#django.test.SimpleTestCase.assertInHTML

The WagtailTestCase also has a BeautifulSoup helper:
`utils.wagtail_tests.WagtailTestUtils.get_soup`.
BS is an awesome package to parse HTML and validate its contents: https://www.crummy.com/software/BeautifulSoup/